commit c203928fae924e017f39afb2723414d458fc01a1
Author: Daiki Ueno <ueno@gnu.org>
Date:   Thu Jul 17 18:38:04 2014 +0900

    Fix test failure regarding int->float conversion
    
    Supply only small integers (< 1^23) to int->float conversion tests,
    so they can roundtrip without error.

diff --git a/test/msgpack_test.cpp b/test/msgpack_test.cpp
index fb35c68..6917caf 100644
--- a/test/msgpack_test.cpp
+++ b/test/msgpack_test.cpp
@@ -201,10 +201,8 @@ TYPED_TEST_P(IntegerToFloatingPointTest, simple_buffer)
   v.push_back(1);
   if (numeric_limits<integer_type>::is_signed) v.push_back(-1);
   else v.push_back(2);
-  v.push_back(numeric_limits<integer_type>::min());
-  v.push_back(numeric_limits<integer_type>::max());
   for (unsigned int i = 0; i < kLoop; i++) {
-    v.push_back(rand());
+    v.push_back(rand() % 0x7FFFFF);
   }
   for (unsigned int i = 0; i < v.size() ; i++) {
     msgpack::sbuffer sbuf;
